src/DataProviders/SqlDataProvider/SqlDataProvider.ts

```diff
@@ -1,61 +1,83 @@
-﻿//import sql from "mssql";
-const sql = require('mssql');
-import * as assert from "assert";
-import { IDataProvider } from "../IDataProvider";
-import { ConnectionPool, ConnectionError, config, IResult } from "mssql";
-import { SeasonInfo } from "../../DataModels/SeasonInfo";
+﻿import * as assert from "assert";
+import { inject, injectable } from "inversify";
+import { ConnectionPool } from "mssql";
+import { isNullOrUndefined } from "util";
+import { Configuration } from "../../Common/Configuration";
+import { SeasonInfo } from "../../Models/SeasonInfo";
+import { TYPES } from "../../types";
+import { IDataProvider, IDataProviderOptions } from "../IDataProvider";
+import { HazzatDbSchema } from "./HazzatDbSchema";
+
+/**
+ * Sql Data Provider Options
+ */
+@injectable()
+export class SqlDataProviderOptions implements IDataProviderOptions {
+    // TODO: Is there a way to pass this in from App?
+    public settings: string = Configuration.dbConnectionString;
+}
 
 /*
  * Sql Data Provider
  */
+@injectable()
 export class SqlDataProvider implements IDataProvider {
     private _tablePrefix: string = "Hymns_";
+    private _cp: Promise<ConnectionPool>;
 
-    constructor(private _config: config) {
+    constructor(
+        @inject(TYPES.IDataProviderOptions) options: IDataProviderOptions
```

Comment:
this is a nit, but usually `Options` classes are not injectable, since they tend to differ by the usage across the app (while usually things that get injected are singletons). If something is going to be used in the same way everywhere in the app, usually that doesn't take an `Options` class, but it might take a singleton Configuration object injected here.

not a big deal, just FYI

Reply:
I'll just have the SqlDataProvider take a dependency on the app configuration.   I didn't really like that but I'll just do it